packages/hadron-build/lib/target.js

```diff
@@ -106,16 +107,32 @@ function getPkg(directory) {
 }
 
 class Target {
+  // eslint-disable-next-line complexity
   constructor(dir, opts = {}) {
     this.dir = dir || process.cwd();
     this.out = path.join(this.dir, 'dist');
 
     const pkg = getPkg(dir);
     this.pkg = pkg;
 
+    const defaultArch =
+      // eslint-disable-next-line no-nested-ternary
+      process.platform === 'darwin'
+        ? os.cpus().some((cpu) => {
+          // process.arch / os.arch() will return the arch for which the node
+          // binary was compiled. Checking if one of the CPUs has Apple in its
```

Comment:
Can we download the right version of `node.js` instead, or additionally to doing this, seems something we would want to do anyway, or there are downsides to doing that?

Reply:
Not while Compass is on Node.js 14, there are no darwin arm binaries for this version

Reply:
That's right!

Reply:
Just to clarify is doing it that way sounds good to you or do you suggest to bump Node.js / Electron version we are using? Because the latter will require resolving some breaking changes first in Compass. Also if you're worried about this auto detecting logic we can just manually pass the arch in the evergreen config and remove this code altogether

Reply:
oh no not at all, the auto-detect part is fine, i forgot that node 14 doesn't have an ARM build

Reply:
After Le Rouxs PR lands we can actually try upgrading I think, and then it will be native arm (and both mongosh and compass will be on the same node version 🎉)